### ISSUE TITLE
[Feature Request] Wrap around in focus and move-through

### ISSUE THREAD

**Issue by flunderpero** [nikitabobko/AeroSpace#23]
It would be nice if they would wrap around to the first or last window respectively.

And by the way: This is absolutely awesome!

**Comment by nikitabobko** [comment]
I agree that it'd be nice to implement the feature for the `focus` command. In i3, it's called "focus wrapping" https://i3wm.org/docs/userguide.html#_focus_wrapping

"move wrapping" for the `move-through` command is an interesting idea, I didn't think about it. Right now, moving a window out of the container either literally moves the window out of the container or creates an implicit container. 

But I suppose it indeed can be made configurable to wrap the movement when the movement is done in a direction that is parallel to the container's orientation (the movement would still "move a window out of the container" if the movement is done in a direction that is perpendicular to the container's orientation)

I need more time to think about "move wrapping". I'm not yet convinced that it will be popular enough to spend resources on implementing and supporting this feature. 
- Other arguments, opinions, and inputs are welcome
- References to tiling window managers that implement this feature would help

**Comment by nikitabobko** [comment]
> the movement would still "move a window out of the container" if the movement is done in a direction that is perpendicular to the container's orientation

Though, it's an open question for both: `focus` and `move-through` commands

**Comment by nikitabobko** [comment]
@flunderpero how did you come up with the idea of "move wrapping"? Do you have any cool use cases? Or maybe it's a feature from some other window manager?

**Comment by nikitabobko** [comment]
> In i3, it's called "focus wrapping" https://i3wm.org/docs/userguide.html#_focus_wrapping

And I don't like i3 `yes|no|force|workspace` names. Every time read them, it takes me time to understand the differences between them. Right now, I'm thinking about something like this:
| i3 name | Theoretical AeroSpace name |
|---------|----------------------------|
| yes | I'm not sure what the name should it be. Right now, I'm rather thinking that it's an inconsistent configuration, so I'd rather not provide it |
| no | off |
| workspace | workspace |
| force | 'container',  'parent', or 'parent_container' |

Examples:
```
focus-wrapping = 'workspace'
focus-wrapping = 'container'
focus-wraping = 'parent_container'
```

**Comment by flunderpero** [comment]
Wrap around for move-through is just stupid, sorry. I don't know why I included it.

I would like to just bind one key-combo to cycle through all the windows in a workspace (I have three at most).

**Comment by flunderpero** [comment]
I like your thoughts on doing this right in all cases. Perhaps `focus.boundaries-action` could be an alternative name, but I think all of your suggestions would work.

**Comment by nikitabobko** [comment]
1. I decided not to implement `container` boundary and `wrap-around-container` boundary action. If I implemented `container` boundary, I'd need to implement `focus child|parent` which I don't want to do (see #5)

2. I implemented the mentioned flags as `focus` command options `focus --boundaries <boundary> --boundaries-action <action>`. I don't want to introduce config options that change the behavior of commands. It would make scripting harder (also see #43 Same reasons)

Please let me know if you disagree

**Comment by nikitabobko** [comment]
Released in 0.7.0-Beta

**Comment by flunderpero** [comment]
@nikitabobko It works great! Thank you!
